scst-store/amr/graphql-query.hbs.md

```diff
@@ -0,0 +1,261 @@
+
+### <a id='location-query-args'></a> Location Query Arguments
+
+Below are the supported arguments that users can specify when querying for `Location`. They are all optional.
+
+- `query`: expects an object that specifies additional arguments used to query, supported arguments in this query object are listed below
+  
+  - `alias`: name of the location, `String` value. By default, a location's `alias` is set to be the same as the `reference`. It is also configurable by the user, see [AMR Configuration](configuration.hbs.md)
+    
+    Example:
+    ```
+    locations(query:{alias: "test-cluster"})
+    ```
+    
+  - `reference`: reference of the location, `String` value. A location's `reference` is set to be the `kube-system` namespace UID.
+  
+    Example:
+    ```
+    locations(query:{reference: "f4f63c6a10ed7cfb06dfb03c2b2d6a9d5bbe95931c71f8cb346fd2284b1e5d82"})
+    ```
+    
+  - `labels`: specify labels the location must contain, expects a list of labels. 
+    A label has `key` and `value` fields of `String` type, at least one of those fields must be provided when querying with labels. 
+  
+    >**Note** The result will return locations that have `all` given labels (i.e. this is an `AND` operation).
+  
+    Example:
+    ```
+    locations(query:{ labels:[{value: "run"}, 
+                              {key: "env"}, 
+                              {key: "region", value: "east1b"} ]
+    })
+    ```
+
+### <a id='location-fields'></a> Location Fields
+
+Below are fields that users can choose to return in the GraphQL query. At least one field needs to be specified.
+
+- `alias`: name of the location
+- `reference`: reference of the location
+- `labels`: labels associated with the location, has the following fields
+  - `key`: key of the label
+  - `value`: value of the label
+  
+### Sample Location Queries
+
+- Query for location with the `alias` `"test-cluster"`. 
+  In the query results, return information on the `reference`, `alias`, and `labels` showing both `key` and `value`.
+  ```
+  query getLocationByAlias {
+    locations(query:{alias: "test-cluster"}) {
+      reference
+      alias
+      labels {
+        key
+        value
+      }
+    }
+  }
+  ```
+  
+- Query for locations that contains two `labels`, one with `key` `"env"` and one `{key: "region", value: "east1b"}`.
+  In the query results, return information on `alias` and `reference` only.
+  ```
+  query getLocationByLabel {
+    locations(query:{labels:[{key: "env"}, 
+                              {key: "region", value: "east1b"} ]
+    }) {
+      reference
+      alias
+    }
+  }
+  ```
+
+## Querying for Apps
+
+This section provides details on the GraphQL query arguments, and the list of fields available for `App`.
+
+### App Query Parameters
+
+Below are the supported arguments that users can specify when querying for `App`, they are all optional.
```

Comment:
⛏️ Consistency with the phrase `, they are all optional.` Above in the "Locations" section you had it as `. They are all optional.`

I'm OK with either version, just need to be consistent.